src/Adapter/TaxRulesGroup/AbstractTaxRulesGroupHandler.php

```diff
@@ -0,0 +1,249 @@
+     * @param TaxRule $taxRule
+     *
+     * @return bool
+     *
+     * @throws CannotDeleteTaxRuleException
+     */
+    protected function deleteTaxRule(TaxRule $taxRule): bool
+    {
+        try {
+            return $taxRule->delete();
+        } catch (PrestaShopException $e) {
+            throw new CannotDeleteTaxRuleException(
+                sprintf('An error occurred when deleting tax rule object with id "%s".', $taxRule->id)
+            );
+        }
+    }
+
+    /**
+     * @param TaxRulesGroup $taxRulesGroup
+     *
+     * @throws TaxRulesGroupConstraintException
+     * @throws PrestaShopException
+     */
+    protected function validateTaxRulesGroupFields(TaxRulesGroup $taxRulesGroup): void
+    {
+        if (!$taxRulesGroup->validateFields(false)) {
+            throw new TaxRulesGroupConstraintException(
+                'Tax rules group contains invalid field values'
+            );
+        }
+    }
+
+    /**
+     * @param TaxRule $taxRule
+     *
+     * @throws PrestaShopException
+     * @throws TaxRuleConstraintException
+     */
+    protected function validateTaxRuleFields(TaxRule $taxRule): void
+    {
+        if (!$taxRule->validateFields(false)) {
+            throw new TaxRuleConstraintException(
+                'Tax rule contains invalid field values'
+            );
+        }
+    }
+
+    /**
+     * @param int $taxRulesGroupId
+     * @param int $countryId
+     * @param int $stateId
+     * @param int|null $taxRuleId
+     *
+     * @return bool
+     */
+    protected function assertUniqueBehaviorTaxRuleForCountry(
+        int $taxRulesGroupId,
+        int $countryId,
+        int $stateId,
+        ?int $taxRuleId = null
+    ): bool {
+        $errors = $this->validator->validate([
+            'taxRulesGroupId' => $taxRulesGroupId,
+            'country' => $countryId,
+            'state' => $stateId,
+            'taxRuleId' => $taxRuleId,
+        ], new UniqueTaxRuleBehavior());
+
+        return 0 !== count($errors);
+    }
+
+    /**
+     * @param string $zipCode
+     * @param int $countryId
+     *
+     * @return bool
+     */
+    protected function assertIsValidZipCode(
+        string $zipCode,
+        int $countryId
+    ): bool {
+        $errors = $this->validator->validate([
+            'zipCode' => $zipCode,
+            'country' => $countryId,
+        ], new ZipCodeRange());
+
+        return 0 !== count($errors);
+    }
+
+    /**
+     * @param CountryDataProvider $countryDataProvider
+     * @param int $langId
+     * @param CountryId|null $countryId
+     *
+     * @return int[]
+     *
+     * @throws CountryNotFoundException
+     */
+    protected function getCountryForTaxRule(
+        CountryDataProvider $countryDataProvider,
+        int $langId,
+        ?CountryId $countryId
+    ): array {
+        $selectedCountries = [];
+
+        if ($countryId !== null && $countryId->getValue() > 0) {
+            $selectedCountries[] = $countryId->getValue();
```

Comment:
```suggestion
            return $countryId->getValue();
```
If I'm not wrong, you can return right now in this case